interoperator/pkg/errors/errors.go

```diff
@@ -0,0 +1,179 @@
+func NewUnmarshalError(message string, err error) *InteroperatorError {
+	return &InteroperatorError{
+		Err:     err,
+		Code:    CodeUnmarshalError,
+		Message: message,
+	}
+}
+
+// IsUnmarshalError checks whether error is of CodeUnmarshalError type
+func IsUnmarshalError(err error) bool {
+	return ErrorCode(err) == CodeUnmarshalError
+}
+
+// NewConvertError returns new error indicating error converting between formats
+func NewConvertError(message string, err error) *InteroperatorError {
+	return &InteroperatorError{
+		Err:     err,
+		Code:    CodeConvertError,
+		Message: message,
+	}
+}
+
+// IsConvertError checks whether error is of CodeConvertError type
+func IsConvertError(err error) bool {
+	return ErrorCode(err) == CodeConvertError
+}
+
+// NewSFServiceNotFound returns a new error which indicates that the SfService is not found.
+func NewSFServiceNotFound(name string, err error) *InteroperatorError {
+	return &InteroperatorError{
+		Err:     err,
+		Code:    CodeSFServiceNotFound,
+		Message: fmt.Sprintf("SfService %s not found", name),
+	}
+}
+
+// IsSFServiceNotFound is true if the error indicates the requested service is not found.
+func IsSFServiceNotFound(err error) bool {
+	return ErrorCode(err) == CodeSFServiceNotFound
+}
+
+// NewSFPlanNotFound returns a new error which indicates that the SfService is not found.
+func NewSFPlanNotFound(name string, err error) *InteroperatorError {
+	return &InteroperatorError{
+		Err:     err,
+		Code:    CodeSFPlanNotFound,
+		Message: fmt.Sprintf("SfPlan %s not found", name),
+	}
+}
+
+// IsSFPlanNotFound is true if the error indicates the requested service plan is not found.
+func IsSFPlanNotFound(err error) bool {
+	return ErrorCode(err) == CodeSFPlanNotFound
+}
+
+// NewSFServiceInstanceNotFound returns a new error which indicates that the SfService is not found.
+func NewSFServiceInstanceNotFound(name string, err error) *InteroperatorError {
+	return &InteroperatorError{
+		Err:     err,
+		Code:    CodeSFServiceInstanceNotFound,
+		Message: fmt.Sprintf("SFServiceInstance %s not found", name),
+	}
+}
+
+// IsSFServiceInstanceNotFound is true if the error indicates the requested service instance is not found.
+func IsSFServiceInstanceNotFound(err error) bool {
+	return ErrorCode(err) == CodeSFServiceInstanceNotFound
+}
+
+// NewSFServiceBindingNotFound returns a new error which indicates that the SfService is not found.
+func NewSFServiceBindingNotFound(name string, err error) *InteroperatorError {
+	return &InteroperatorError{
+		Err:     err,
+		Code:    CodeSFServiceBindingNotFound,
+		Message: fmt.Sprintf("SFServiceBinding %s not found", name),
+	}
+}
+
+// IsSFServiceBindingNotFound is true if the error indicates the requested service binding is not found.
+func IsSFServiceBindingNotFound(err error) bool {
+	return ErrorCode(err) == CodeSFServiceBindingNotFound
+}
+
+// IsNotFound is true if the error indicates any not found error
+func IsNotFound(err error) bool {
```

Comment:
`IsNotFound ` => `NotFound `? And for all similar error types also if it makes sense